### ISSUE TITLE
Culture-specific shape alternates

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/Orchard#6879]
We could easily add alternates to each shapes for every site culture (so they're visible in shape tracing; by fetching from `ICultureManager.ListCultures()`) or for the current culture (so it's faster to add, not having to retrieve cultures).

E.g. the `User` shape now would have a `User.en-US` alternate too.

**Comment by Piedone** [comment]
@Jetski5822 since you worked a lot on localization, what do you think?

**Comment by Skrypt** [comment]
The localizationPart "polymorphs" the shape already ? Does everything needs to be localizable ?

**Comment by Jetski5822** [comment]
We don have this already!?!?!? Wow - I missed that!

ListCultures() is cached... so if you have en-US as a culture you would have

User
User.en
User.en-US

As I think i always provides the generic one (might be wrong but if I am, you should have that too)

**Comment by Piedone** [comment]
@Skrypt I think having culture-specific alternates is orthogonal to localizing content items, the two complement each other. Thus I wouldn't tie producing such alternates to content items being localized with LocalizationPart (given that there can be other ways of content localization any way).

@Jetski5822 it was strange that we don't have this for me too! Indeed `ListCultures()` is cached so no need to worry about this.

**Comment by Jetski5822** [comment]
It should be a feature you enable.... btw :)

**Comment by Piedone** [comment]
Yes, good point.

**Comment by Skrypt** [comment]
Yes somehow there is something "breaking" the composition design pattern since the culture is also a global setting. I just mean, that if we would need to use this on custom content types, I would use at least the localizationPart for knowing if I should add the option in the shape tracers. I don't know about performance though on that. Using the cached list might be faster :wink:
